database-commons/src/main/java/io/cdap/plugin/db/batch/action/AbstractDBArgumentSetter.java

```diff
@@ -0,0 +1,141 @@
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.db.batch.action;
+
+import io.cdap.cdap.etl.api.FailureCollector;
+import io.cdap.cdap.etl.api.PipelineConfigurer;
+import io.cdap.cdap.etl.api.StageConfigurer;
+import io.cdap.cdap.etl.api.action.Action;
+import io.cdap.cdap.etl.api.action.ActionContext;
+import io.cdap.cdap.etl.api.action.SettableArguments;
+import io.cdap.plugin.db.ConnectionConfig;
+import io.cdap.plugin.util.DBUtils;
+import io.cdap.plugin.util.DriverCleanup;
+import java.sql.Connection;
+import java.sql.Driver;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Properties;
+
+/**
+ * Action that converts db column into pipeline argument.
+ */
+public class AbstractDBArgumentSetter extends Action {
+
+  private static final String JDBC_PLUGIN_ID = "driver";
+  private final ArgumentSetterConfig config;
+
+  public AbstractDBArgumentSetter(ArgumentSetterConfig config) {
+    this.config = config;
+  }
+
+  @Override
+  public void run(ActionContext context) throws Exception {
+    Class<? extends Driver> driverClass = context.loadPluginClass(JDBC_PLUGIN_ID);
+    FailureCollector failureCollector = context.getFailureCollector();
+    SettableArguments settableArguments = context.getArguments();
+    processArguments(driverClass, failureCollector, settableArguments);
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer)
+      throws IllegalArgumentException {
+    DBUtils.validateJDBCPluginPipeline(pipelineConfigurer, config, JDBC_PLUGIN_ID);
+    Class<? extends Driver> driverClass = DBUtils.getDriverClass(
+        pipelineConfigurer, config, ConnectionConfig.JDBC_PLUGIN_TYPE);
+    StageConfigurer stageConfigurer = pipelineConfigurer.getStageConfigurer();
+    FailureCollector collector = stageConfigurer.getFailureCollector();
+    config.validate(collector);
+    try {
+      processArguments(driverClass, collector, null);
+    } catch (SQLException e) {
+      collector.addFailure("SQL error while executing query: " + e.getMessage(), null)
+          .withStacktrace(e.getStackTrace());
+    } catch (IllegalAccessException | InstantiationException e) {
+      collector.addFailure("Unable to instantiate JDBC driver: " + e.getMessage(), null)
+          .withStacktrace(e.getStackTrace());
+    } catch (Exception e) {
+      collector.addFailure(e.getMessage(), null).withStacktrace(e.getStackTrace());
+    }
+    config.validate(collector);
```

Comment:
why do we need to call `config.validate(collector)` 2 times?

Reply:
@CuriousVini removed redundant call